R/run_mcmc_diagnostics.R

```diff
@@ -0,0 +1,187 @@
+#' Run MCMC using {adnuts} for SS3
+#'
+#' @param dir_wd A file path to the directory where the model files and the
+#'   executable are saved. This can be relative to your current working
+#'   directory or a full path. Your working directory will be changed while
+#'   executing the function to be inside the location provided in this
+#'   argument but upon exiting it will return to your current working
+#'   directory.
+#' @param model A string giving the name of the executable without an
+#'   an extension. The extension is provided in `extension`. Do not include
+#'   any path information, just the name of the executable. For example, the
+#'   default is `"ss3"`.
+#' @param extension A string giving the extension for the executable name
+#'   provided in `model`. The default is `.exe`, which works on Windows
+#'   machines.
+#' @param iter A numeric value for the number of MCMC draws to do. Default is
+#'   2000.
+#' @param chains A numeric value for the number of chains to run. Default is 2.
+#' @param interactive A logical, where `TRUE` will run
+#'   [adnuts::launch_shinyadmb()]. The default is `FALSE`.
+#' @param verbose A logical, specifying if information should be printed
+#'   to the screen. The default is `FALSE` but using `TRUE` can be helpful
+#'   when you are first running the function or when you have a very unstable
+#'   model.
+#'
+#' @export
+#' @author Cole M. Monnahan (cole.monnahan@noaa.gov; February 2021) and
+#'   augmented by Kelli F. Johnson
+#'
+run_mcmc_diagnostics <- function(
+  dir_wd,
+  model = "ss3",
+  extension = ".exe",
+  iter = 2000,
+  chains = 2,
+  interactive = FALSE,
+  verbose = FALSE) {
+  # Set up and test model for running. This requires
+  # pointing to a folder and executable. The folder needs to
+  # contain all sufficient input files and assumes optimization
+  # has occurred and produced all necessary outputs. Temporary
+  # copies will be made in the working directory during execution
+  wd <- getwd()
+  on.exit(setwd(wd), add = TRUE)
+  stopifnot(file.exists(dir_wd))
+  setwd(dir_wd)
+
+  # Check for exe
+  exe_present <- file.exists(dir_wd, paste0(model, extension))
+  if (!exe_present) {
+    cli::cli_abort(
+      "An executable called {paste0(model, extension)} was not found in the directory. The
+      executable is required to be in the folder for this function."
+    )
+  }
+
+  # Run the model if need be
+  #if (!file.exists("Report.sso")) {
+  #  system(paste(model, "-nohess"))
+  #}
+
+  # Regularization
+  p <- "_mcmc"
+  dir.create(p, showWarnings = FALSE)
+  dir.create(file.path(p, "fits"))
+  r4ss::copy_SS_inputs(
+    dir.old = getwd(),
+    dir.new = p,
+    overwrite = TRUE,
+    verbose = verbose
+  )
+  file.copy(
+    paste0(model, extension),
+    file.path(p, paste0(model, extension))
+  )
+
+  # optimize w/ -mcmc flag b/c of bias adjustment.
+  setwd(p)
```

Comment:
@chantelwetzel-noaa, thank you for all the improvements!
One trivial additional change is that you can now remove the `setwd(p)` and `setwd("..")` because `p` is passed to `r4ss::run()`.